common/backintime.py

```diff
@@ -1144,6 +1174,113 @@ def snapshotsListPath(args):
         _umount(cfg)
     sys.exit(RETURN_OK)
 
+def lastSnapshotDict(config):
+    """
+    Return a dictionary containing the profile name and most recent snapshot
+    data for the given profile id. If the snapshot was unsuccessful, the
+    date of the most recent successful snapshot is included in the dict.
+
+    Args:
+        config: 
+                        The currently selected configuration.
+
+    Returns:
+        dict:
+                        the most recent successful snapshot for the profile.
+    """
+
+    try:
+        errors = snapshots.lastSnapshot(config).failed
+        info = {_('Last snapshot'):
+            str(snapshots.lastSnapshot(config).date),
+                _('Successful'): not errors}
+        if errors:
+            last_success = None
+            for snapshot in snapshots.listSnapshots(config):
+                if not snapshot.failed:
+                    last_success = snapshot.date
+                    break
+            info.update({_('Last successful'):
+                str(last_success) if last_success else None})
+        return {config.profileName(config.currentProfile()): info}
+    except:
+        return {config.profileName(config.currentProfile()):
+                {_('Last snapshot'): None}}
+
+def profileStatus(args):
+    """
+    Print most recent snapshot and snapshot details for the profile
+    specified with --profile or --profile-id (--profile takes 
+    preference).
+
+    Args:
+        args (argparse.Namespace):
+                        previously parsed arguments
+
+    Raises:
+        SystemExit:     0
+    """
+    force_stdout = setQuiet(args)
+    cfg = getConfig(args)
+    _mount(cfg)
+    if args.profile:
+        if not cfg.setCurrentProfileByName(args.profile):
+            _umount(cfg)
+            logger.error(args.profile)
+    elif not cfg.setCurrentProfile(args.profile_id):
+        _umount(cfg)
+        logger.error(args.profile_id)
+    id = cfg.currentProfile()
+    info = lastSnapshotDict(cfg)
+    info[cfg.profileName(id)].update({
+            _('Paths'): {
+                _('Snapshots'): cfg.snapshotsFullPath(),
+                _('Path exists'): cfg.canBackup(),
+                _('Mount point'): cfg.get_snapshots_mountpoint(),
+                _('Log file'): cfg.takeSnapshotLogFile(),
+            }})
+    print(json.dumps(info, indent=2), file=force_stdout)
+    _umount(cfg)
+    sys.exit(RETURN_OK)
```

Comment:
Don't be scared of blank lines. ;)

```suggestion
    force_stdout = setQuiet(args)
    cfg = getConfig(args)
    _mount(cfg)

    if args.profile:
        if not cfg.setCurrentProfileByName(args.profile):
            _umount(cfg)
            logger.error(args.profile)

    elif not cfg.setCurrentProfile(args.profile_id):
        _umount(cfg)
        logger.error(args.profile_id)

    id = cfg.currentProfile()
    info = lastSnapshotDict(cfg)
    info[cfg.profileName(id)].update({
            _('Paths'): {
                _('Snapshots'): cfg.snapshotsFullPath(),
                _('Path exists'): cfg.canBackup(),
                _('Mount point'): cfg.get_snapshots_mountpoint(),
                _('Log file'): cfg.takeSnapshotLogFile(),
            }})

    print(json.dumps(info, indent=2), file=force_stdout)
    _umount(cfg)

    sys.exit(RETURN_OK)
```